src/compiler/checker.ts

```diff
@@ -37683,8 +37683,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             { flags: FlowFlags.Start };
         const trueCondition: FlowCondition = {
             flags: FlowFlags.TrueCondition,
-            node: expr,
             antecedent,
+            node: expr,

```

Comment:
I tried to follow the structure used by the ones created in the binder to keep functions receiving this monomorphic. I'm not sure if my change prompted [this further experiment](https://github.com/microsoft/TypeScript/pull/57977) but it certainly looked like it to me since it was open soon after I opened this one here ;p

TLDR: this was meant as an optimization for the engine. I didn't measure it at all since I don't have time for it but it's usually better to create structures in the same way (order of keys etc matters for the engine)

Reply:
👍 Makes sense. The calls to `getFlowTypeOfReference` are so expensive compared to everything else in `getTypePredicateFromBody` that I doubt micro-optimizations matter (see https://github.com/microsoft/TypeScript/pull/57465#issuecomment-1974921974 and https://github.com/microsoft/TypeScript/pull/57660). But I can't imagine they hurt!

Reply:
once u apply micro-optimizations at scale there are some good wins to be had :P https://github.com/microsoft/TypeScript/pull/57977#issuecomment-2032268884